src/preproc/preprocessing_pass_registry.cpp

```diff
@@ -0,0 +1,52 @@
+/*********************                                                        */
+/*! \file preprocessing_pass_registry.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Justin Xu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Preprocessing pass registry
+ **
+ ** Preprocessing pass registry that registers the preprocessing passes.
+ **/
+#include "preproc/preprocessing_pass_registry.h"
+#include <utility>
+#include "base/output.h"
+#include "preproc/preprocessing_pass.h"
+
+namespace CVC4 {
+namespace preproc {
+
+void PreprocessingPassRegistry::init(PreprocessingPassAPI* api) {
+  for (std::pair<std::string, PreprocessingPass*>&& element :
+       d_stringToPreprocessingPass) {
+    element.second->init(api);
+  }
+}
+
+void PreprocessingPassRegistry::registerPass(
+    const std::string& ppName, PreprocessingPass* preprocessingPass) {
+  Debug("pp-registry") << "Registering pass " << ppName << std::endl;
```

Comment:
We should have an assert here that checks that no pass with the same name has already been registered.